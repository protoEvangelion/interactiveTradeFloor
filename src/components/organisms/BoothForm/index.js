import React, { Component, PropTypes } from 'react'
import { Field } from 'redux-form'
import styled from 'styled-components'

import { ReduxField, Button } from 'components'

const Form = styled.form`
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
  margin-top: -20px;
`

class BoothForm extends Component {
  constructor() {
    super()
    this.state = { email: 'false' }
  }
  render() {
    const { handleSubmit, email, submitting, company, description, owner, status } = this.props
    return (
      <Form method="POST" onSubmit={handleSubmit}>
        <Field name="_csrf" type="hidden" component="input" />
        <Field name="company" label="Company Name" defaultValue={company} component={ReduxField} />
        <Field name="owner" label="Owner" component={ReduxField} type="select" value={owner}>
          <option value="None">None</option>
          <option value="Todd">Todd</option>
          <option value="Richard">Richard</option>
          <option value="Jin">Jin</option>
        </Field>
        <Field name="status" label="Status" component={ReduxField} type="select" value={status}>
          <option value="n/a">N/A</option>
          <option value="open">Open</option>
          <option value="holding">Holding</option>
          <option value="collect">Need to Collect</option>
          <option value="good">Good to go</option>
        </Field>
        <Field name="description" label="Description" component={ReduxField} defaultValue={description} />
        <br />
        <div>
          <Button style={{ marginRight: '10px' }} type="submit" disabled={submitting}>Save</Button>
          <Button type="submit" disabled={submitting} palette="secondary" onClick={email}>Save & Email Team</Button>
        </div>
      </Form>
    )
  }
}

BoothForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  email: PropTypes.func.isRequired,
  submitting: PropTypes.bool,
  boothNum: PropTypes.number.isRequired,
  company: PropTypes.string.isRequired,
  description: PropTypes.string,
  owner: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
}

export default BoothForm
