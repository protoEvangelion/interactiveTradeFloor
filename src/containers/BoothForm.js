import React from 'react'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { fromForm } from 'store/selectors'
import { postCreateRequest } from 'store/actions'
import { createValidator, required } from 'services/validation'

import { BoothForm } from 'components'

const BoothFormContainer = props => <BoothForm {...props} />

const onSubmit = (data, dispatch) => new Promise((resolve, reject) => {
  dispatch(postCreateRequest(data, resolve, reject))
})

const validate = createValidator({
  title: [required],
  body: [required],
})

const mapStateToProps = (state) => ({
  initialValues: {
    _csrf: fromForm.getCsrfToken(state),
  },
})

export const config = {
  form: 'BoothForm',
  fields: ['title', 'body'],
  destroyOnUnmount: false,
  onSubmit,
  validate,
}

export default connect(mapStateToProps)(reduxForm(config)(BoothFormContainer))
