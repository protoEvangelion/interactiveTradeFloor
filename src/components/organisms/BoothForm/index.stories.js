import React from 'react'
import { storiesOf } from '@kadira/storybook'
import { BoothForm } from 'containers'

storiesOf('BoothForm', module)
  .add('default', () => (
    <BoothForm />
  ))
