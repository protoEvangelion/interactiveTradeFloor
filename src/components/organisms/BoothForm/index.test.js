import React from 'react'
import { shallow } from 'enzyme'
import BoothForm from '.'

const handleSubmit = jest.fn()

const wrap = (props = {}) => shallow(<BoothForm handleSubmit={handleSubmit} {...props} />)

it('calls renderSubmit when submitted', () => {
  handleSubmit.mockClear()
  const wrapper = wrap()
  expect(handleSubmit).not.toBeCalled()
  wrapper.simulate('submit')
  expect(handleSubmit).toBeCalled()
})

it('disables button while submitting', () => {
  const wrapper = wrap()
  expect(wrapper.find({ disabled: true }).length).toBe(0)
  wrapper.setProps({ submitting: true })
  expect(wrapper.find({ disabled: true })).toHaveLength(1)
})
