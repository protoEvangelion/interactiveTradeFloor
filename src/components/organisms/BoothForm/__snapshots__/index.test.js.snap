// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoothForm snapshot should render the snapshot 1`] = `
<form
  className="edkfuk"
  method="POST"
  onSubmit={[Function]}
>
  <input
    name="_csrf"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="hidden"
    value=""
  />
  <span />
  <span>
    <option
      value="None"
    >
      None
    </option>
    <option
      value="Todd"
    >
      Todd
    </option>
    <option
      value="Richard"
    >
      Richard
    </option>
    <option
      value="Jin"
    >
      Jin
    </option>
  </span>
  <span>
    <option
      value="n/a"
    >
      N/A
    </option>
    <option
      value="open"
    >
      Open
    </option>
    <option
      value="holding"
    >
      Holding
    </option>
    <option
      value="collect"
    >
      Need to Collect
    </option>
    <option
      value="good"
    >
      Good to go
    </option>
  </span>
  <span />
  <br />
  <div>
    <span>
      Save
    </span>
    <span>
      Save & Email Team
    </span>
  </div>
</form>
`;
